atlasdb-config/src/main/java/com/palantir/atlasdb/config/ShouldRunBackgroundSweepSupplier.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.config;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.sweep.queue.config.TargetedSweepInstallConfig;
+import com.palantir.atlasdb.sweep.queue.config.TargetedSweepRuntimeConfig;
+
+/**
+ * Synthesises provided install and runtime configurations, and uses this to decide whether the background sweeper
+ * should run. Specifically:
+ *
+ * - if the background sweeper has been explicitly enabled or disabled, use that setting;
+ * - otherwise if Targeted Sweep is fully enabled (both writing to the sweep queue and actually sweeping), disable
+ *   background sweep
+ * - otherwise, follow {@link AtlasDbConstants#DEFAULT_ENABLE_SWEEP}.
```

Comment:
I think we want to be defensive here, mainly because for products in the ecosystem of large internal product they're likely to not use TS and I can't be certain that all of them explicitly set BS to enabled as opposed to relying on it being the default.

This can be changed later on once TS has broader adoption internally, but I think is a separate change that I wouldn't want to make yet